riverlog/river_log.go

```diff
@@ -0,0 +1,149 @@
+// Package riverlog provides a context logging middleware for workers that
+// collates output and stores it to job records.
+package riverlog
+
+import (
+	"bytes"
+	"cmp"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"log/slog"
+
+	"github.com/riverqueue/river/internal/jobexecutor"
+	"github.com/riverqueue/river/rivershared/baseservice"
+	"github.com/riverqueue/river/rivertype"
+)
+
+const (
+	maxSizeMB    = 2
+	maxSizeBytes = maxSizeMB * 1024 * 1024
+	metadataKey  = "river:log"
```

Comment:
@bgentry Oh yeah I forgot to ask about this one. I put a "namespace" in here because I feel like something like "log" or "logs" it too generic and risks collision with things that users might've been doing.

This could also be `river_log` or something like that too. The colon is kind of XML inspired I guess. It occurred to me that maybe we should've namespaced the "job output" key as well for the same reason. Regardless, we should probably have some convention for future additions.

Thoughts?

Reply:
Yeah I saw, I don't have a strong opinion on what it should be tbh. So far I think I was thinking that for metadata we could assume it's safe to claim any top-level plain key and then tell users _they_ should namespace if using metadata; most won't touch it so this would be pretty safe advice. But not a strong preference.